### PR TITLE
Adds isort and codespell to pre-commit

### PR DESCRIPTION
**Pull request by CyclingNinja**

### CODE REVIEW

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from .version import version as __version__
 
-{%- if cookiecutter.include_example_code == 'y' %}
 from .example_mod import do_primes
+{%- if cookiecutter.include_example_code == 'y' %}
 # Then you can be explicit to control what ends up in the namespace,
```
Comment (nabobalis) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py]: Going to need to make sure isort doesnt move this with # isort:skip ?

Comment (CyclingNinja) [reply {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py]: yeah this is what i'm trying to sort atm :disappointed:

File: {{ cookiecutter.package_name }}/docs/conf.py
Comment (Cadair) [inline {{ cookiecutter.package_name }}/docs/conf.py:1]: upstream

File: {{ cookiecutter.package_name }}/setup.py
Comment (Cadair) [inline {{ cookiecutter.package_name }}/setup.py:1]: upstream

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py
Comment (Cadair) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py:1]: upstream

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/example_mod.py
Comment (Cadair) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/example_mod.py:1]: upstream

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
-{%- if cookiecutter.include_example_code == 'y' %}
 
-from .example_mod import do_primes
+{% if cookiecutter.include_example_code == 'y' -%}
 # Then you can be explicit to control what ends up in the namespace,
 __all__ = ['do_primes']
-{% else %}
+{%- else -%}
 __all__ = []
 {%- endif -%}
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py]:
```suggestion
{% endif -%}
```

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+{%- endif %}
 from .version import version as __version__
-{%- if cookiecutter.include_example_code == 'y' %}
 
-from .example_mod import do_primes
+{% if cookiecutter.include_example_code == 'y' -%}
 # Then you can be explicit to control what ends up in the namespace,
 __all__ = ['do_primes']
-{% else %}
+{%- else -%}
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py]:
```suggestion
{% else -%}
```

File: {{ cookiecutter.package_name }}/.pre-commit-config.yaml [hunk 4]
```diff
@@ -20,6 +25,10 @@ repos:
         exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*|.json)$|^CITATION.rst$"
       - id: mixed-line-ending
         exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*)$"
-ci:
-  autofix_prs: false
-  autoupdate_schedule: "quarterly"
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/.pre-commit-config.yaml]: You don't want to remove this.

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -14,3 +14,5 @@ repos:
         exclude: "{{ cookiecutter.package_name }}"
       - id: debug-statements
         exclude: "{{ cookiecutter.package_name }}"
+ci:
+  autofix_prs: false
```
Comment (nabobalis) [inline .pre-commit-config.yaml:18]: Do we want to add the auto update?

Comment (Cadair) [reply .pre-commit-config.yaml:18]: It's on by default, so this should be fine?

File: {{ cookiecutter.package_name }}/.pre-commit-config.yaml [hunk 6]
```diff
@@ -5,6 +5,11 @@ repos:
     hooks:
       - id: ruff
         args: ["--fix"]
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*|extern.*|sunpy/extern)$"
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/.pre-commit-config.yaml]:
```suggestion
        exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*|extern.*|{{ cookiecutter.module_name }}/extern)$"
```

File: {{ cookiecutter.package_name }}/.pre-commit-config.yaml [hunk 7]
```diff
@@ -20,6 +25,13 @@ repos:
         exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*|.json)$|^CITATION.rst$"
       - id: mixed-line-ending
         exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*)$"
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.6
+    hooks:
+      - id: codespell
+        additional_dependencies:
+          - tomli
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/.pre-commit-config.yaml]: As our config is no longer in pyproject.toml presumably this isn't needed? @nabobalis 
```suggestion
```

**Comment by Cadair** [comment]
Merged with 3.9 failing as we realised that 3.9 is past it's SPEC 0 expiration date and will be removed soon.

**Pull request merged**